src/dbnode/persist/fs/commitlog/writer.go

```diff
@@ -379,9 +382,15 @@ func (w *fsChunkWriter) Write(p []byte) (int, error) {
 
 	// Write contents to file descriptor
 	n, err := w.fd.Write(w.buff)
+	// Count bytes successfully written from slice p
+	pBytesWritten := n - chunkHeaderLen
```

Comment:
As discussed--this fix makes sense to me as well, since I think we're now correctly implementing the `Writer` interface (meaning that bufio won't choke on it). ccing @robskillington as well though, since I know it was different than your suggestion (and obviously you have a helluva lot more context on the code :) )

Reply:
Yeah that sounds good, if we've tested it and it expects as behaves now with this I'm onboard with it (and looks like it does).

Reply:
Yeah. This change has been unit tested and cluster tested by @justinjc